worlds/__init__.py

```diff
@@ -95,27 +100,94 @@ def load(self) -> bool:
             return False
 
 
-# find potential world containers, currently folders and zip-importable .apworld's
-world_sources: List[WorldSource] = []
-for folder in (folder for folder in (user_folder, local_folder) if folder):
-    relative = folder == local_folder
-    for entry in os.scandir(folder):
-        # prevent loading of __pycache__ and allow _* for non-world folders, disable files/folders starting with "."
-        if not entry.name.startswith(("_", ".")):
-            file_name = entry.name if relative else os.path.join(folder, entry.name)
-            if entry.is_dir():
-                world_sources.append(WorldSource(file_name, relative=relative))
-            elif entry.is_file() and entry.name.endswith(".apworld"):
-                world_sources.append(WorldSource(file_name, is_zip=True, relative=relative))
-
-# import all submodules to trigger AutoWorldRegister
-world_sources.sort()
-for world_source in world_sources:
-    world_source.load()
-
-# Build the data package for each game.
-from .AutoWorld import AutoWorldRegister
-
-network_data_package: DataPackage = {
-    "games": {world_name: world.get_data_package_data() for world_name, world in AutoWorldRegister.world_types.items()},
-}
+def scan_worlds() -> List[WorldSource]:
+    # find potential world containers, currently folders and zip-importable .apworld's
+    sources: List[WorldSource] = []
+    for folder in (folder for folder in (user_folder, local_folder) if folder):
+        relative = folder == local_folder
+        for entry in os.scandir(folder):
+            # prevent loading of __pycache__ and allow _* for non-world folders, disable files/folders starting with "."
+            if not entry.name.startswith(("_", ".")):
+                file_name = entry.name if relative else os.path.join(folder, entry.name)
+                if entry.is_dir():
+                    sources.append(WorldSource(file_name, relative=relative))
+                elif entry.is_file() and entry.name.endswith(".apworld"):
+                    sources.append(WorldSource(file_name, is_zip=True, relative=relative))
+    sources.sort()
+    return sources
+
+
+def get_world_by_name(name: str) -> str:
+    for world in world_data:
+        if world["game"] == name:
```

Comment:
The typing of `get_worlds_info` says that `world_data` is `Dict[str, Dict[str, str]]`
which means `world` here would be `str`
which means `world["game"]` would crash.